guvnor-m2repo-editor/guvnor-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/helpers/HttpPostHelper.java

```diff
@@ -208,9 +207,13 @@ private String uploadPom( final FormData formData ) throws IOException {
                                    version );
                 }
             } catch (ProjectBuildingException pbe) {
-                return pbe.getMessage();
+                log.error(pbe.getMessage(),
+                          pbe);
+                return UPLOAD_UNABLE_TO_PARSE_POM;
             } catch (MavenEmbedderException mee) {
-                return mee.getMessage();
+                log.error(mee.getMessage(),
+                          mee);
+                return UPLOAD_UNABLE_TO_PARSE_POM;
             }catch (Exception e) {
                 return UPLOAD_UNABLE_TO_PARSE_POM;
             }
```

Comment:
Is there a specific reason to _not_ log the general `Exception`? If not, the whole catch block could be simplified to just
```
catch (Exception ex) {
    log.error("Can no parse the provided POM", ex); // or some better log message
    return UPLOAD_UNABLE_TO_PARSE_POM;
}
```
which also brings different question: should we even catch these general `Exception`s? Shouldn't they be propaged and caught the the top-level `UncaughtExceptionHandler` (or how we call it)?